services/httpd/handler.go

```diff
@@ -1087,13 +1089,23 @@ func gzipFilter(inner http.Handler) http.Handler {
 			inner.ServeHTTP(w, r)
 			return
 		}
-		gz := gzip.NewWriter(w)
-		defer gz.Close()
+		gz := gzipWriterPool.Get().(*gzip.Writer)
+		gz.Reset(w)
+		defer func() {
+			gz.Close()
+			gzipWriterPool.Put(gz)
```

Comment:
similar to [here](https://github.com/influxdata/influxdb/blob/91ee34b111f2034ec9456a6ce391b470d69e3216/tsdb/engine/tsm1/pools.go), but instead calling `Reset` with the new `ResponseWriter` on `get`, and possibly even calling `Close()` in the `put` function.

The new code in the handler could then be

```go
gz := getGzipWriter(w)
defer putGzipWriter(gz)
```